adminbot.py

```diff
@@ -47,17 +56,33 @@ def call_help(self, op):
         # this is going to be handled by a listener plugin
         pass
 
+@auth.verify_password
+def verify_password(username, password):
+    if username in USERS:
+        return check_password_hash(USERS.get(username), password)
+    return False
+
+with open('pw.secret') as p:
+    pw = p.read().strip()
+with open('channels.json') as c:
+    channels = json.load(c)
+ops = [
+    'vilmibm',
+    'archangelic',
+    'equa',
+    'l0010o0001l'
+]
+bot = AdminBot(channels, 'adminbot', 'localhost', ops=ops, ns_pass=pw)
+
+@app.route('/notify', methods=['POST'])
+@auth.login_required
+def notify_admins():
+    channel = request.json['channel']
+    message = request.json['message']
+    bot.connection.privmsg(channel, message)
 
 if __name__ == '__main__':
-    with open('pw.secret') as p:
-        pw = p.read().strip()
-    with open('channels.json') as c:
-        channels = json.load(c)
-    ops = [
-        'vilmibm',
-        'archangelic',
-        'equa',
-        'l0010o0001l'
-    ]
-    bot = AdminBot(channels, 'adminbot', 'localhost', ops=ops, ns_pass=pw)
-    bot.start()
+    bot_thread = threading.Thread(target=bot.start)
+    bot_thread.start()
+    flask_thread = threading.Thread(target=app.run)
```

Comment:
i think this is a very reasonable way to run flask given that this is (for now) a low volume API that isn't exposed to the world.